src/Nest/CommonOptions/TimeUnit/Time.cs

```diff
@@ -20,7 +21,7 @@ public class Time : IComparable<Time>, IEquatable<Time>
 		private const double MicrosecondsInAMillisecond = 10;
 
 		private static readonly Regex ExpressionRegex =
-			new Regex(@"^(?<factor>[-+]?\d+(?:\.\d+)?)\s*(?<interval>(?:y|w|d|h|m|s|ms|nanos|micros))?$",
+			new Regex(@"^(?<factor>[-+]?[\d\.e\-]+?)\s*(?<interval>(?:y|w|d|h|m|s|ms|nanos|micros))?$",
```

Comment:
@russcam merging this in, if you have more cases for the regex to cover let me know here 👍 this was fun 😄